src/interactivity/win32/ut_interactivity_win32/UiaTextRangeTests.cpp

```diff
@@ -1136,57 +1136,82 @@ class UiaTextRangeTests
         // the UTR should refuse to move past the end.
 
         const auto bufferSize{ _pTextBuffer->GetSize() };
-        const til::point endInclusive = { bufferSize.RightInclusive(), bufferSize.BottomInclusive() };
-        const auto endExclusive{ bufferSize.EndExclusive() };
+        const COORD origin{ bufferSize.Origin() };
+        const COORD lastLineStart{ bufferSize.Left(), bufferSize.BottomInclusive() };
+        const COORD secondToLastCharacterPos{ bufferSize.RightInclusive() - 1, bufferSize.BottomInclusive() };
+        const COORD endInclusive{ bufferSize.RightInclusive(), bufferSize.BottomInclusive() };
+        const COORD endExclusive{ bufferSize.EndExclusive() };
 
-        // Iterate over each TextUnit. If the we don't support
+        // Iterate over each TextUnit. If we don't support
         // the given TextUnit, we're supposed to fallback
         // to the last one that was defined anyways.
+        BEGIN_TEST_METHOD_PROPERTIES()
+            TEST_METHOD_PROPERTY(L"Data:textUnit", L"{0, 1, 2, 3, 4, 5, 6}")
+            TEST_METHOD_PROPERTY(L"Data:degenerate", L"{false, true}")
+        END_TEST_METHOD_PROPERTIES();
+
+        int unit;
+        bool degenerate;
+        VERIFY_SUCCEEDED(TestData::TryGetValue(L"textUnit", unit), L"Get TextUnit variant");
+        VERIFY_SUCCEEDED(TestData::TryGetValue(L"degenerate", degenerate), L"Get degenerate variant");
+        TextUnit textUnit{ static_cast<TextUnit>(unit) };
+
         Microsoft::WRL::ComPtr<UiaTextRange> utr;
         int moveAmt;
-        for (int unit = TextUnit::TextUnit_Character; unit != TextUnit::TextUnit_Document; ++unit)
-        {
-            Log::Comment(NoThrowString().Format(L"Forward by %s", toString(static_cast<TextUnit>(unit))));
+        Log::Comment(NoThrowString().Format(L"Forward by %s", toString(textUnit)));
 
-            // Create a degenerate UTR at EndExclusive
+        // Create an UTR at EndExclusive
+        if (degenerate)
+        {
+            THROW_IF_FAILED(Microsoft::WRL::MakeAndInitialize<UiaTextRange>(&utr, _pUiaData, &_dummyProvider, endExclusive, endExclusive));
+        }
+        else
+        {
             THROW_IF_FAILED(Microsoft::WRL::MakeAndInitialize<UiaTextRange>(&utr, _pUiaData, &_dummyProvider, endInclusive, endExclusive));
-            THROW_IF_FAILED(utr->Move(static_cast<TextUnit>(unit), 1, &moveAmt));
-
-            VERIFY_ARE_EQUAL(endExclusive, utr->_end);
-            VERIFY_ARE_EQUAL(0, moveAmt);
         }
+        THROW_IF_FAILED(utr->Move(textUnit, 1, &moveAmt));
+
+        VERIFY_ARE_EQUAL(endExclusive, utr->_end);

```

Comment:
so moving forward from a degenerate _or_ non-degenerate range results in a degenerate range? great

Reply:
Yup. That functionality was caught and introduced by #7530. So this does more testing on that for free!